src/lib/engine/index.ts

```diff
@@ -58,7 +58,19 @@ export function runGame(code: string, canvas: HTMLCanvasElement, onPageError: (e
 	code = `"use strict";\n${code}`
 	
 	try {
-		parseScript(code)
+		const program = parseScript(code, { loc: true })
+		for (let item of program.body) {
+			if (item.type === "FunctionDeclaration" && Object.keys(api).includes(item.id!.name)) {
```

Comment:
This might be auto-optimized, but would suggest keeping the API keys precached just once on startup instead of potentially recreating them repeatedly in this loop.